Source/Tests/LinuxSyscalls/Syscalls.h

```diff
@@ -349,6 +349,38 @@ uint64_t CloneHandler(FEXCore::Core::CpuStateFrame *Frame, FEX::HLE::clone3_args
     return flags;
   }
 
+/**
+ * @brief Checks raw syscall return for error
+ *
+ * This should only be used with raw syscall usage
+ *
+ * This should not be used with glibc wrapped syscall functions
+ *   - This includes the glibc ::syscall(...) function
+ *   - This is due to glibc already wrapping the return and setting errno
+ *
+ * This function should not be used with UAPI breaking syscall results
+ * ioctl specifically will break this convention.
+ *
+ * @param Result The raw syscall return
+ *
+ * @return If the result was an error result
+ */
+
+[[maybe_unused]]
```

Comment:
If `[[maybe_unused]]` is just here due to the compiler yelling about `HasSyscallError` not being used in all .cpp files that include `Syscalls.h`, I think `(static) inline bool HasSyscallError` should work without that attribute.

Reply:
Yes, and the warning stays around with YouCompleteMe sadly. Only thing that resolved it was `[[maybe_unused]]`